src/tracer.cc

```diff
@@ -404,6 +437,10 @@ TraceManager::Trace::StartSpan(
   if (parent_id == 0 && activity == TRITONSERVER_TRACE_REQUEST_START) {
     parent_span_key = kRootSpan;
   } else if (activity == TRITONSERVER_TRACE_REQUEST_START) {
+    // [FIXME] For BLS requests parent span for children's request spans
+    // should be parent model's compute span. Currently,
+    // this won't work, since parent's compute span will be created
+    // only after children's spans are created.
```

Comment:
Current issue does not have a major impact on OpenTelemetry traces. All child models are still displayed under the parrent's model.  So unless the correct order is requested, we can keep it as is.

Regarding your suggestion, I think it would require a major refactor on trace files to do that. I don't think we need to implement it now.